src/scalar.rs

```diff
@@ -459,6 +462,28 @@ impl Default for Scalar {
     }
 }
 
+impl TryFrom<&[u8]> for Scalar {
+    type Error = CurveError;
+
+    fn try_from(bytes: &[u8]) -> Result<Self, Self::Error> {
+        if bytes.len() != 32 {
+            return Err(CurveError(
+                InternalError::BytesLengthError{name: "Scalar", length: 32}));
+        }
+
+        let mut s_bytes = [0x00u8; 32];
+        s_bytes.copy_from_slice(bytes);
+
+        Ok(Scalar::from_bits(s_bytes))
```

Comment:
I feel very nervous calling `from_bits()` under the hood here, it should only be called when explicitly needed, for protocols that rely on it such as x25519 diffie-hellman, because it allows unreduced `Scalar`s to be constructed.